src/main/java/bio/terra/service/profile/ProfileService.java

```diff
@@ -303,4 +304,17 @@ public void verifyDeployedApplication(
               + "operation");
     }
   }
+
+  public void registerManagedResourceGroup(
+      BillingProfileRequestModel request, AuthenticatedUserRequest user) {
+    String billingProfileId = request.getBillingAccountId();
+    ManagedResourceGroupCoordinates managedResourceGroupCoordinates =
+        new ManagedResourceGroupCoordinates()
+            .tenantId(request.getTenantId().toString())
+            .subscriptionId(request.getSubscriptionId().toString())
+            .managedResourceGroupName(request.getResourceGroupName());
```

Comment:
I had left this remark in the ticket:

> The format of the MRG coordinates that TDR billing profile creation expects from the caller may not match exactly the format expected by the Sam endpoint.

I was specifically wondering if TDR's `BillingProfileRequestModel.resourceGroupName` is the expected value for `managedResourceGroupName`.  In TDR Azure architecture, the user creates a Resource Group in which to deploy their TDR managed app.  The TDR managed app manages a new Managed Resource Group.  I think TDR billing profile creation takes in the user-created Resource Group name, but the Sam endpoint takes in Managed Resource Group name.

Reply:
I am not really sure how to address this issue. The `request` doesn't seem to contain the "real" Managed Resource Group name.

Reply:
The Azure service has the following:

```
var properties = objectMapper.valueToTree(applicationDeployment.properties());
var managedResourceGroupId = properties.get(MANAGED_RESOURCE_GROUP_ID_KEY)
```

I'm not sure if this is safe to access from the running steps, however.

Reply:
I think you are on the right track.  The request body may not contain the MRG name, but this is something TDR can obtain through interactions with the Azure SDK… and the snippet you found shows that there is precedence elsewhere in the code for doing just that.

(BTW -- where possible to link to snippets from the existing codebase, that would be much appreciated to make sure that we're looking at the same thing, and leave a trail of breadcrumbs for those perusing this PR in the future.)

The snippet you shared looks to come from `AzureApplicationDeploymentService.newApplicationDeployment` which is called only in `AzureApplicationDeploymentService.getOrRegisterApplicationDeployment`.  

https://github.com/DataBiosphere/jade-data-repo/blob/3198570b711a39510684a03b0cd685756d839b6b/src/main/java/bio/terra/service/resourcemanagement/azure/AzureApplicationDeploymentService.java#L88

`getOrRegisterApplicationDeployment` is presently only called when creating datasets and snapshots, as part of getting or creating their Azure storage accounts.

I may be missing historical reasoning, but it makes more sense to me to register an application deployment as part of creating an Azure billing profile, as a step within the flight that can also delete the application deployment registration in its `undoStep` definition.  Then the remainder of the flight can rely on having the MRG name available.

Any future attempts to create billing profiles from the same MRG would fail fast at billing profile creation rather than later on at dataset creation, which seems to be the desired effect:

https://github.com/DataBiosphere/jade-data-repo/blob/3198570b711a39510684a03b0cd685756d839b6b/src/main/java/bio/terra/service/resourcemanagement/azure/AzureApplicationDeploymentService.java#L75-L81